projects/soft_999_eddy_kim_fw_103/src/main.c

```diff
@@ -0,0 +1,49 @@
+#include "adc.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+#include "wait.h"
+
+static void callback(const GpioAddress *address, void *context) {
+  // Is it right to pass the GpioAddress A6 by *context? I'm unsure what *address is supposed to
+  // return
+  GpioAddress *A6 = context;
+  uint16_t A6_data = 0;
+  adc_read_raw_pin(*A6, &A6_data);
+  LOG_DEBUG("%d\n", A6_data);
+  return;
+}
+
+int main(void) {
+  gpio_init();
+  interrupt_init();
+
+  // Unsure if these are defined properly
+  GpioAddress B2 = { .port = GPIO_PORT_B, .pin = 2 };
+  GpioAddress A6 = { .port = GPIO_PORT_A, .pin = 6 };
+
+  // Unsure if these are defined properly as well
+  GpioSettings B2_settings = { GPIO_DIR_IN, GPIO_STATE_LOW, GPIO_RES_NONE, GPIO_ALTFN_ANALOG };
+
+  GpioSettings A6_settings = { GPIO_DIR_IN, GPIO_STATE_LOW, GPIO_RES_NONE, GPIO_ALTFN_ANALOG };
```

Comment:
Two points about this:
* This is correct for A6, `GPIO_ALTFN_ANALOG` is the correct alt-function (something different for the MCU to do with the pin) for reading from ADC. For B2, you don't need to read from ADC for it, so use `GPIO_ALTFN_NONE` instead.
* On a stylistic note, we usually prefer GNU-style struct initialization syntax like this:
```c
GpioSettings settings = {
  .direction = GPIO_DIR_IN,
  .state = GPIO_STATE_LOW,
  .resistor = GPIO_RES_NONE,
  .alt_function = GPIO_ALTFN_ANALOG,
};
```
That way it's easy to verify that the fields match up with the values since it's not dependent on ordering. (Also, the names `B2_settings` and `A6_settings` don't *really* comply with the coding style since they start with capitals.)

Reply:
Whoops, I should pay more attention to the coding standards...